packages/yarnpkg-core/sources/Workspace.ts

```diff
@@ -157,6 +157,40 @@ export class Workspace {
     return workspaceList;
   }
 
+  /**
+   * Find workspaces which include the current workspace as a dependency/devDependency recursively.
+   *
+   * @param rootWorkspace root workspace
+   * @param project project
+   *
+   * @returns all the workspaces marked as dependents
+   */
+  getRecursiveWorkspaceDependents({dependencies = Manifest.hardDependencies}: {dependencies?: Array<HardDependencies>} = {}) {
+    const workspaceList = new Set<Workspace>();
+
+    const visitWorkspace = (workspace: Workspace) => {
+      this.project.workspaces.filter(projectWorkspace => {
+        for (const dependencyType of dependencies) {
+          for (const descriptor of projectWorkspace.manifest[dependencyType].values()) {
+            const foundWorkspace = this.project.tryWorkspaceByDescriptor(descriptor);
+            if (foundWorkspace !== null && structUtils.areLocatorsEqual(foundWorkspace.locator, workspace.locator)) {
```

Comment:
@arcanis is this the right way to check workspace equality?
I'm also not super clear on the difference between the `locator` and `anchoredLocator`.